pyMOE/gdsconverter.py

```diff
@@ -0,0 +1,298 @@
+        """
+        Creates the gds layout using raster mode where each data point is a pixel rectangle to
+        be defined in the layout
+        
+        """
+        self.gdslib = gdspy.GdsLibrary()
+        
+        cell = gdspy.Cell(cellname,exclude_from_current=True)
+        
+        self.layers = np.arange(len(self.levels))
+        total_layers = len(self.layers)
+        total_points = self.mask.shape[0]*self.mask.shape[1]
+        
+        size_x, size_y = self.mask.shape
+        XX = self.mask.XX
+        YY = self.mask.YY
+
+        half_pixel_x = self.mask.pixel_x/2
+        half_pixel_y = self.mask.pixel_y/2
+
+        # normalize to units:
+        XX = XX/self.units
+        YY = YY/self.units
+        half_pixel_x = half_pixel_x/self.units
+        half_pixel_y = half_pixel_y/self.units
+        datatype = 0
+        
+        # Creates top cell that will reference the remaining cells
+        topcell = gdspy.Cell(cellname, exclude_from_current=True)
+    
+        # Make evaluation of computational effort
+        if self.verbose:
+            print("Mask has %d number of points distributed in %d layers"%(total_points, total_layers))       
+
+        with Timer("Total time converting to GDS"):
+            # Creates a list of cells where each cell will correspond to a layer, to enable merging afterwards.
+            list_cells = []
+            for layer in self.layers:
+                cellname = "tempcelllayer%d"%(layer)
+
+                cell = gdspy.Cell(cellname,exclude_from_current=True)
+                list_cells.append(cell)
+
+            # Iterates to create a layout in each layer corresponding to one discretized level
+            if self.verbose:
+                print("Creating individual pixel polygons")
+            with Timer("Create Polygons"):
+                for i in range(size_x):
+                    for j in range(size_y):
+
+                        current_point = i*size_x+j
+                        aperture_value = self.aperture[i,j]
+
+                        current_layer = aperture_value
+                        x = XX[i,j]
+                        y = YY[i,j]
+                        
+                        # Creates corners of rectangle with center at the data value
+                        rectangle_first_corner = (x-half_pixel_x,y-half_pixel_y)
+                        rectangle_second_corner = (x+half_pixel_x,y+half_pixel_y)
+                        
+                        #Creates rectangle and adds it to the cell corresponding to the current layer
+                        rect = gdspy.Rectangle(rectangle_first_corner, rectangle_second_corner, current_layer, datatype)
+                        cell = list_cells[current_layer]
+                        
+                        # Saves each rectangle to a separate cell so we can merge more easily afterwards
+                        cell.add(rect)
```

Comment:
Several options to improve:
1) Memory - print the cells to file after each sizey iteration avoiding keeping the memory occupied 
2) Memory and speed - use gds instatiation functionality

Reply:
1. To be added as issue. Can check export module function grayim2gds_writer for an example of implementation of gdspy which uses the gdspy writer. 

2. To be investigated with gdspy, suggest to add as issue - pya merge function can also merge instances, a comparison between instantiation+merge and non-instantiation+merge (merge using pya library) is shown at the end of notebook   https://github.com/INLnano/pyMOE/blob/dev-refactor-compatibility/notebooks/3%20-%20Import_export/Import-export.ipynb